tools/edgeDeviceFiles.sh

```diff
@@ -142,12 +154,28 @@ function getClusterName () {
 	echo ""
 }
 
-# Create a IBM Cloud Private platform API key
+# Check if an IBM Cloud Pak platform API key exists
+function checkAPIKey () {
+	echo "Checking if API key already exists for $USER..."
+	echo "cloudctl iam api-keys | cut -d' ' -f4 | grep \"$USER-API-Key\""
+
+	KEY=$(cloudctl iam api-keys | cut -d' ' -f4 | grep "$USER-API-Key")
+	if [ -z $KEY ]; then
+		echo "\"$USER-API-Key\" does not exist. A new one will be created."
+        echo ""
+        CREATE_NEW_KEY=true
+    else 
+    	CREATE_NEW_KEY=false
+    fi
+    echo ""
+}
+
+# Create a IBM Cloud Pak platform API key
 function createAPIKey () {
-	echo "Creating IBM Cloud Private platform API key..."
-	echo "cloudctl iam api-key-create \"$EDGE_DEVICE API Key\" -d \"$EDGE_DEVICE API Key\" -f key.txt"
+	echo "Creating IBM Cloud Pak platform API key..."
+	echo "cloudctl iam api-key-create \"$USER-API-Key\" -d \"$USER API Key\" -f key.txt"
 
-	cloudctl iam api-key-create "$EDGE_DEVICE API Key" -d "$EDGE_DEVICE API Key" -f key.txt
+	cloudctl iam api-key-create "$USER-API-Key" -d "$USER API Key" -f key.txt
```

Comment:
I suggest the key name be a little more unique, e.g. `$USER-Edge-Device-API-Key`

Reply:
Changed